src/instructlab/config/init.py

```diff
@@ -183,6 +202,59 @@ def init(
     )
 
 
+def hw_auto_detect() -> tuple[str | None, int | None, int | None, _train, bool]:
+    # Third Party
+    import torch
+
+    profiles = render_profiles()
+    gpus = 0
+    total_vram = 0
+    train = _train()
+    chosen_vram = None
+    chosen_profile_name = None
+    previous_vram = None
+    # try nvidia
+    if torch.cuda.is_available():
+        click.echo("Detecting Hardware...")
+        gpus = torch.cuda.device_count()
+        for i in range(gpus):
+            properties = torch.cuda.get_device_properties(i)
+            total_vram += properties.total_memory  # memory in B
+    vram = convert_bytes_to_proper_mag(total_vram)[0]
```

Comment:
It's not a problem now but there is a risk of future bug here in that you are comparing the output of a routine that uses dynamic units with a routine that uses fixed units (render_profiles). This function convert_bytes_to_proper_mag() is a perfect printed output function, and could be extended to return TB, at which point a system with >= TB of vram would look like 1GB in the loop below.  There is also a less serious more of a nit issue that a float is being compared to an int.  I would recommend either just comparing in bytes, or if you want to downscale the unit convert it to an int of the floor of GB

Reply:
Cool. Yes, the recently released H200 systems are available for use/purchase have > 1TB (1128 GB x 8 - (141GB per card)). We definitely want to run on those.